core/src/main/java/io/substrait/expression/proto/ExpressionProtoConverter.java

```diff
@@ -465,57 +471,68 @@ public Expression visit(io.substrait.expression.Expression.Window expr) throws R
+      return offsetIsPositive(expr.value());
+    }
+
+    @Override
+    public Long visit(io.substrait.expression.Expression.I16Literal expr) throws RuntimeException {
+      return offsetIsPositive(expr.value());
+    }
+
+    @Override
+    public Long visit(io.substrait.expression.Expression.I32Literal expr) throws RuntimeException {
+      return offsetIsPositive(expr.value());
+    }
+
+    @Override
+    public Long visit(io.substrait.expression.Expression.I64Literal expr) throws RuntimeException {
+      return offsetIsPositive(expr.value());
+    }
+  }
+
   private Expression.WindowFunction.Bound toBound(io.substrait.expression.WindowBound windowBound) {
     var boundedKind = windowBound.boundedKind();
-    Expression.WindowFunction.Bound bound = null;
-    switch (boundedKind) {
-      case CURRENT_ROW -> bound =
-          Expression.WindowFunction.Bound.newBuilder()
-              .setCurrentRow(Expression.WindowFunction.Bound.CurrentRow.getDefaultInstance())
-              .build();
+    return switch (boundedKind) {
+      case CURRENT_ROW -> Expression.WindowFunction.Bound.newBuilder()
+          .setCurrentRow(Expression.WindowFunction.Bound.CurrentRow.getDefaultInstance())
+          .build();
       case BOUNDED -> {
         WindowBound.BoundedWindowBound boundedWindowBound =
             (WindowBound.BoundedWindowBound) windowBound;
-        var offset = boundedWindowBound.offset();
-        boolean isPreceding = boundedWindowBound.direction() == WindowBound.Direction.PRECEDING;
-        io.substrait.expression.Expression.I32Literal offsetLiteral =
-            (io.substrait.expression.Expression.I32Literal) offset;
-        var offsetVal = offsetLiteral.value();
-        var boundedProto = Expression.WindowFunction.Bound.Unbounded.getDefaultInstance();
-        if (isPreceding) {
-          var offsetProto =
-              Expression.WindowFunction.Bound.Preceding.newBuilder().setOffset(offsetVal).build();
-          bound = Expression.WindowFunction.Bound.newBuilder().setPreceding(offsetProto).build();
-        } else {
-          var offsetProto =
-              Expression.WindowFunction.Bound.Following.newBuilder().setOffset(offsetVal).build();
-          bound = Expression.WindowFunction.Bound.newBuilder().setFollowing(offsetProto).build();
-        }
-      }
-      case UNBOUNDED -> {
-        WindowBound.UnboundedWindowBound unboundedWindowBound =
-            (WindowBound.UnboundedWindowBound) windowBound;
-        boolean isPreceding = unboundedWindowBound.direction() == WindowBound.Direction.PRECEDING;
-        var unboundedProto = Expression.WindowFunction.Bound.Unbounded.getDefaultInstance();
-        if (isPreceding) {
-          var preceding = Expression.WindowFunction.Bound.Preceding.newBuilder().build();
-          bound =
-              Expression.WindowFunction.Bound.newBuilder()
-                  .setUnbounded(unboundedProto)
-                  .setPreceding(preceding)
```

Comment:
This was problematic for two reasons:
1. `unbounded` and `proceeding` are both part of the same one of, so both should not be set.
2. The default offset for `preceding` is set to 0 here, which is dissalowed per the spec (offset must be a positive integer). I noticed this issue when I added the `LiteralToWindowBoundOffset` converter which checked for the positivity of the offset.